cmd/os-readdir_unix.go

```diff
@@ -143,6 +143,10 @@ func readDirN(dirPath string, count int) (entries []string, err error) {
 	bufp := direntPool.Get().(*[]byte)
 	defer direntPool.Put(bufp)
 
+	nameTmp := direntPool.Get().(*[]byte)
```

Comment:
I think we need to initialize another pool and use it here.. direntPool contains a list of directory entries, so better to have a separate pool.

Reply:
It is a byte slice pool - it works fine for this. Better to share pools than having it separate and we know it is big enough.

Reply:
> It is a byte slice pool - it works fine for this

The thing is that the dirent pool is configurable, currently it generates 8k bytes.. but it can be reduced in the future (it was 1kb before), which is not a good thing if you use it when parsing file names

Reply:
Well, there should always be room for the file name, otherwise I wouldn't expect the syscall to work in the first place?

Reply:
We will never change the byte slice below 8192 because that is the maximum possible length for a file name across all platforms.

The actual buffer is 100 times that so I think it is okay to reuse the same buffer